samples/v1/types_config.go

```diff
@@ -0,0 +1,212 @@
+package v1
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Config contains the configuration and detailed condition status for the Samples Operator.
+type Config struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ConfigSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
```

Comment:
needs godoc - ex: `// spec is the desired configuration of the Samples Operator`

Reply:
yeah I went back and forth on this ... some of the existing examples in openshift/api seemed to eschew comments here in favor of the comments in the types below (i.e. ConfigSpec in this case).

see 
- https://github.com/openshift/api/blob/master/console/v1/types_console_cli_download.go
- https://github.com/openshift/api/blob/master/operator/v1/types_authentication.go
- https://github.com/openshift/api/blob/master/operator/v1/types_openshiftapiserver.go

and many other @adambkaplan 

I'd rather stay consistent with the pattern established elsewhere.

You OK with that ?

Reply:
There's plenty of other counter-examples. At this point if someone complains about the content in `oc explain` they can file a BZ.

Reply:
True (which is why I went back and forth) ... in the end, looking at the generated CRD yaml, there are descriptions for ConfigSpec/ConfigStatus so I expect we will be OK